src/ComputeSharp.D2D1/Shaders/Interop/Effects/PixelShaderEffect.Properties.cs

```diff
@@ -0,0 +1,259 @@
+    /// <param name="actualSize">A optional pointer to a variable that stores the actual number of bytes retrieved on the property.</param>
+    /// <returns>The <see cref="HRESULT"/> for the operation.</returns>
+    private int GetResourceTextureManagerAtIndex(int index, byte* data, uint dataSize, uint* actualSize)
+    {
+        if (!IsResourceTextureManagerIndexValid(index, out _))
+        {
+            return E.E_INVALIDARG;
+        }
+
+        if (data is null)
+        {
+            return E.E_POINTER;
+        }
+
+        if (dataSize < sizeof(void*))
+        {
+            return E.E_INVALIDARG;
+        }
+
+        using ComPtr<ID2D1ResourceTextureManager> resourceTextureManager = this.resourceTextureManagerBuffer[index];
+
+        resourceTextureManager.CopyTo((ID2D1ResourceTextureManager**)data);
+
+        if (actualSize is not null)
+        {
+            *actualSize = (uint)sizeof(void*);
+        }
+
+        return S.S_OK;
+    }
+
+    /// <summary>
+    /// Sets the resource texture manager for a given index.
+    /// </summary>
+    /// <param name="index">The index of the resource texture manager to set.</param>
+    /// <param name="data">A pointer to the data to be set on the property.</param>
+    /// <param name="dataSize">The number of bytes in the property set by the function.</param>
+    /// <returns>The <see cref="HRESULT"/> for the operation.</returns>
+    private int SetResourceTextureManagerAtIndex(int index, byte* data, uint dataSize)
+    {
+        if (!IsResourceTextureManagerIndexValid(index, out uint dimensions))
+        {
+            return E.E_INVALIDARG;
+        }
+
+        if (data is null)
+        {
+            return E.E_POINTER;
+        }
+
+        void* value = *(void**)data;
+
+        if (value is null)
+        {
+            return E.E_POINTER;
+        }
+
+        if (dataSize != (uint)sizeof(void*))
+        {
+            return E.E_INVALIDARG;
+        }
+
+        using ComPtr<IUnknown> unknown = (IUnknown*)value;
+        using ComPtr<ID2D1ResourceTextureManager> resourceTextureManager = default;
+
+        // Check that the input object implements ID2D1ResourceTextureManager
+        int result = unknown.CopyTo(resourceTextureManager.GetAddressOf());
+
+        if (result != S.S_OK)
+        {
+            return result;
+        }
+
+        using ComPtr<ID2D1ResourceTextureManagerInternal> resourceTextureManagerInternal = default;
+
+        // Then, also check that it implements ID2D1ResourceTextureManagerInternal
+        result = unknown.CopyTo(resourceTextureManagerInternal.GetAddressOf());
+
+        if (result != S.S_OK)
+        {
+            return result;
+        }
+
+        // Initialize the resource texture manager, if an effect context is available
+        if (this.d2D1EffectContext is not null)
+        {
+            resourceTextureManagerInternal.Get()->Initialize(this.d2D1EffectContext, &dimensions);
```

Comment:
you're not checking the HRESULT here?

Reply:
Whoops, good catch! Fixed in ecb28e23826be0cf4467d6b52f3021771192631c 😄